cli

```diff
@@ -1,60 +1,62 @@
-    .boolean('http-always-get')
-    .describe('http-always-get', 'Use always HTTP GET requests, by default HEAD is used for pages without any anchors')
-    .boolean('warn-name-attr')
-    .describe('warn-name-attr', 'show warning if name attribute instead of id was used for an anchor')
-    .string('http-cache')
-    .describe('http-cache', "Directory to store the non failing HTTP responses. If none is specified responses won't be cached.")
-    .default('http-cache-max-age', '1w')
-    .describe('http-cache-max-age', 'Invalidate the cache after the given period. Allowed values: https://www.npmjs.com/package/ms')
-    .help('h')
-    .alias('h', 'help')
-    .argv
+		.example('$0 path/to/html/files', 'checks directory with HTMLfiles for broken links and anchors')
+		.boolean('allow-hash-href')
+		.describe('allow-hash-href', 'If `true`, ignores the `href` `#`')
+		.boolean('disable-external')
+		.describe('disable-external', 'disable checks HTTP links')
+		.boolean('external-only')
+		.describe('external-only', 'check HTTP links only')
+		.array('file-ignore')
+		.describe('file-ignore', 'RegExp to ignore files to scan')
+		.array('url-ignore')
+		.describe('url-ignore', 'RegExp to ignore URLs')
+		.array('url-swap')
+		.describe('url-swap', 'RegExp for URLs which can be replaced on the fly')
+		.boolean('limit-scope')
+		.describe('limit-scope', 'forbid to follow URLs which are out of provided path, like ../somewhere')
+		.boolean('mkdocs')
+		.describe('mkdocs', 'transforming URLS from foo/#bar to foo/index.html#bar')
+		.boolean('javadoc')
+		.describe('javadoc', 'Enable special URL transforming which allows to check iframe deeplinks for local javadoc and scaladoc')
+		.array('javadoc-external')
+		.describe('javadoc-external', 'Domain or base URL to do URL transformation to check iframe deeplinks')
+		.array('http-status-ignore')
+		.describe('http-status-ignore', 'pass HTTP status code which will be ignore, by default only 2xx are allowed')
+		.boolean('json')
+		.describe('json', 'print errors as JSON')
+		.default('http-redirects', 0)
+		.describe('http-redirects', 'Amount of allowed HTTP redirects')
+		.default('http-timeout', 5000)
+		.describe('http-timeout', 'HTTP timeout in milliseconds')
+		.boolean('http-always-get')
+		.describe('http-always-get', 'Use always HTTP GET requests, by default HEAD is used for pages without any anchors')
+		.boolean('warn-name-attr')
+		.describe('warn-name-attr', 'show warning if name attribute instead of id was used for an anchor')
+		.string('http-cache')
+		.describe('http-cache', "Directory to store the non failing HTTP responses. If none is specified responses won't be cached.")
+		.default('http-cache-max-age', '1w')
+		.describe('http-cache-max-age', 'Invalidate the cache after the given period. Allowed values: https://www.npmjs.com/package/ms')
+		.default(`ignore-error-exit`, false)
+		.describe('ignore-error-exit', 'Do not generate a non-zero exit code when errors are encountered')
```

Comment:
It totally makes sense to allow to not exit the tool when some errors do occur.

The description could be simplified: we don't need to mention non-zero, we actually want to not exist at all, right?

And I would even simplify the flag to `ignore-error`

Reply:
the link checker will always exit: if it finished, it has an exit code 0, so what we want here is for that exit code to _still_ be 0 even if there were errors, so that build chains that half on non-zero exit codes keep running =)

Reply:
Yes I agree and of course it will exit always. What I meant was that it won't exit prematurely.

Maybe `no-exit-on-error` is more clear.

Reply:
ah yes that might be a better name, I'll see about updating the PR as soon as I have some free time.

Reply:
hm, looking at this again, I think the original description is correct though: my need was to have link-checker run but if there were errors in any of the links, not exist with a code>0: it was already running to completion, checking every ULR, but in order not to break an `npm` chain the exit code _had_ to be zero even if the number of links with errors was greater than zero..

Reply:
Ah okay, if it was running already before to completion, we can maybe use `--ignore-exit-code". And maybe we can simplify in the description instead of negate twice something like this: `Do always exit with a zero status code, even if errors are found`.

Reply:
I can change it back to `ignore exit code` but it's not really an ignore (it's not ignoring its own exit code). I like the new phrasing, what about `--clean-exit` with `Always exit with status code zero, even if errors are found`?

Reply:
I understand what you mean with clean but I don't think that is good terminology in the context of exit codes.

The flag should describe it in a simple way, like `--ignore-exit-code`, `--force-zero-exit-code` or something like that.

Reply:
I'll change it to `--force-zero-exit-code` in that case, since that feels the most correct wrt the actual behaviour.

Reply:
Apologies for the radio silence, life was not kind for a while. I've updated the PR to reflect the new flag.